crewai_tools/tools/stagehand_tool/example.py

```diff
@@ -0,0 +1,119 @@
+        "in the `url` parameter along with the instruction (e.g., instruction='Go to Google', url='https://google.com').\n"
+        "- Use 'extract' when the goal is to pull structured data from the page. Provide a clear `instruction` "
+        "describing what data to extract (e.g., 'Extract all product names and prices').\n"
+        "- Use 'observe' to identify and analyze elements on the current page based on an `instruction` "
+        "(e.g., 'Find all images in the main content area').\n\n"
+        "Remember to break down complex tasks into simple, sequential steps in your `instruction`. For example, "
+        "instead of 'Search for OpenAI on Google and click the first result', use multiple steps with the tool:\n"
+        "1. Use 'navigate' with url='https://google.com'.\n"
+        "2. Use 'act' with instruction='Type OpenAI in the search bar'.\n"
+        "3. Use 'act' with instruction='Click the search button'.\n"
+        "4. Use 'act' with instruction='Click the first search result link for OpenAI'.\n\n"
+        "Always be precise in your instructions and choose the most appropriate command and parameters (`instruction`, `url`, `command_type`, `selector`) for the task at hand."
+    ),
+    llm="gpt-4o",
+    verbose=True,
+    allow_delegation=False,
+    tools=[stagehand_tool],
+)
+
+
+# Define a research task that demonstrates all three primitives
+research_task = Task(
+    description=(
+        "Demonstrate Stagehand capabilities by performing the following steps:\n"
+        "1. Go to https://www.stagehand.dev\n"
+        "2. Extract all the text content from the page\n"
+        "3. Find the Docs link and click on it\n"
+        "4. Go to https://httpbin.org/forms/post and observe what elements are available on the page\n"
+        "5. Provide a summary of what you learned about using these different commands"
+    ),
+    expected_output=(
+        "A demonstration of all three Stagehand primitives (act, extract, observe) "
+        "with examples of how each was used and what information was gathered."
+    ),
+    agent=researcher,
+)
+
+# Alternative task: Real research using the primitives
+web_research_task = Task(
+    description=(
+        "Go to google.com and search for 'Stagehand'.\n"
+        "Then extract the first search result."
+    ),
+    expected_output=(
+        "A summary report about Stagehand's capabilities and pricing, demonstrating how "
+        "the different primitives can be used together for effective web research."
+    ),
+    agent=researcher,
+)
+
+# Set up the crew
+crew = Crew(
+    agents=[researcher],
+    tasks=[research_task],  # You can switch this to web_research_task if you prefer
+    verbose=True,
+    process=Process.sequential,
+)
+
+# Run the crew and get the result
+result = crew.kickoff()
+
+print("\n==== RESULTS ====\n")
+print(result)
+
+# Make sure to clean up stagehand resources
+stagehand_tool.close()
```

Comment:
What do you think about adding contextmanager support or recommending a try/finally approach instead?

We did something similar in MCPServer.